### DISCUSSION TITLE
Async view inspection in combination with evironment

### DISCUSSION THREAD

**Discussion by adusak** [nalexn/ViewInspector#354] [Q&A] [answered]
Hello! I am trying to use the library in combination with Swift Testing and async functions. It works great for simple views, but I am having trouble using views that rely on the `@Environment`. I was trying to find a way, but I have had no luck so far.

```swift
struct MyView: View, in {
    @Environment(\.test) var test: Test
    let inspection = Inspection<Self>()
    var body: some View {
         Text("stuff")
    }
```

```swift
let sut = MyView()
try await ViewHosting.host(sut.environment(\.test, Test()) { hostedView in
    try await hostedView.inspection.inspect { view in
        #expect(throws: Never.self) {
            _ = try view.find(viewWithAccessibilityIdentifier: a11yIdEmptyJobListText)
            _ = try view.find(text: "empty text")
        }
    }
}
```
But then the problem is that the environment view modifier erases the type of `MyView`, and I get an error that `hostedView` does not have a member `inspection`.

Did anybody manage to work around this problem?

**Comment by nalexn** [comment, marked as answer]
Hey, yeah, I found this isn't properly documented, so you can do

```swift
let sut = MyView()
try await ViewHosting.host(sut.environment(\.test, Test()) { _ in
    try await sut.inspection.inspect { view in
        #expect(throws: Never.self) {
            _ = try view.find(viewWithAccessibilityIdentifier: a11yIdEmptyJobListText)
            _ = try view.find(text: "empty text")
        }
    }
}
```

**Reply by adusak** [reply]
Works perfectly! Thank you for the response!

**Reply by nalexn** [reply]
It appears that there is really no point in supplying the view in the closure of `ViewHosting.host`, you can always refer to the outbound `sut` struct for inspection. I'm going to deprecate the existing `host` function and replace its' `whileHosted` closure with `() -> Void`

**Reply by jverkoey** [reply]
Is the provided view not necessary for screenshot testing the hosted view, e.g. as mentioned in https://github.com/nalexn/ViewInspector/issues/306?
